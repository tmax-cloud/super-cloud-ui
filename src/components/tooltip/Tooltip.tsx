import * as React from 'react';
import { styled, Tooltip as MuiTooltip, TooltipProps as MuiTooltipProps, tooltipClasses, SxProps, Theme } from '@mui/material';
import ThemeWrapper from '../../themes/ThemeWrapper';

const StyledTooltip = styled(({ className, ...props }: MuiTooltipProps) => <MuiTooltip {...props} arrow classes={{ popper: className }} />)(({ theme }) => ({
  [`& .${tooltipClasses.arrow}`]: {
    color: theme.palette.tooltip.contentBg,
  },
  [`& .${tooltipClasses.tooltip}`]: {
    backgroundColor: theme.palette.tooltip.contentBg,
    padding: `${theme.spaces.tooltip.contentPaddingTop} ${theme.spaces.tooltip.contentPaddingRight} ${theme.spaces.tooltip.contentPaddingBottom} ${theme.spaces.tooltip.contentPaddingLeft}`,
    fontSize: theme.typography.tooltip.fontSize,
    borderRadius: 0,
    top: -15,
    maxWidth: '18.75rem',
  },
}));

export const Tooltip = (props: TooltipProps) => {
  const { children, open, title, ...rest } = props;
  return (
    <ThemeWrapper>
      <StyledTooltip open={open} title={title} placement="top" arrow {...rest}>
        <span>{children}</span>
      </StyledTooltip>
    </ThemeWrapper>
  );
};

export interface BaseProps {
  /**
   * Tooltip reference element.
   */
  children: React.ReactElement<any, any>;
  /**
   * Callback fired when the component requests to be closed.
   *
   * @param {React.SyntheticEvent} event The event source of the callback.
   */
  onClose?: (event: React.SyntheticEvent | Event) => void;
  /**
   * Callback fired when the component requests to be open.
   *
   * @param {React.SyntheticEvent} event The event source of the callback.
   */
  onOpen?: (event: React.SyntheticEvent) => void;
  /**
   * If `true`, the component is shown.
   */
  open?: boolean;
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx?: SxProps<Theme>;
  /**
   * Tooltip title. Zero-length titles string are never displayed.
   */
  title: NonNullable<React.ReactNode>;
}

export type TooltipProps = BaseProps & React.HTMLAttributes<HTMLDivElement>;
