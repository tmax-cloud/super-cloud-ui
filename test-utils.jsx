// import React from 'react';
// import { render } from '@testing-library/react';
// 추후에 redux 추가하게 되면 redux Provider로 감싸주는 custom render 메서드 추가 해야됨.

export * from '@testing-library/react';
