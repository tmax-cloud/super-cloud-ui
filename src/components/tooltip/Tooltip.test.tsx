import * as React from 'react';
import { render, screen } from '../../../test-utils';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
  test('tooltip renders correctly', () => {
    const { container } = render(
      <Tooltip title="Tooltip Label" open>
        <div>Hover me</div>
      </Tooltip>,
    );
    expect(container).toMatchSnapshot();
  });
  test('tooltip text renders correctly', () => {
    render(
      <Tooltip title="Tooltip Label" open>
        <div>Hover me</div>
      </Tooltip>,
    );
    expect(screen.getByText('Tooltip Label')).toBeInTheDocument();
  });
});
