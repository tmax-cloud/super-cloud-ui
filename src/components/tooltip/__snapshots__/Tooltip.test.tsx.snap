// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip tooltip renders correctly 1`] = `
<div>
  <span
    aria-label="Tooltip Label"
    class=""
    data-mui-internal-clone-element="true"
  >
    <div>
      Hover me
    </div>
  </span>
</div>
`;
